### PR TITLE
key_client: supply GetRandomBytes API.

### PR DESCRIPTION
**Pull request by LykxSassinator**
Signed-off-by: Lucasliang <nkcs_lykx@hotmail.com>

In this pr:
* ~~The `ManagedHsmClient` has been supplied for accessing managed HSM resources.~~
* ~~Supply `GetRandomBytes` API to `ManagedHsmClient` for generating random values.~~
* Supply `GetRandomBytes` API to `KeyClient`.

### CODE REVIEW

**Review by heaths — CHANGES_REQUESTED** [review]
This should be added to the `KeyClient` reflecting what we do in other languages. Don't validate REST methods apart from any URI path parameters, and let the service return a service error as implemented; thus, `KeyClient` would return an error reflecting the service's HTTP 400.

**Comment by LykxSassinator** [comment]
> This should be added to the `KeyClient` reflecting what we do in other languages. 

Several things are a little bit wired to me, triggering me to split `GetRandomBytes` from `KeyClient`:
* The `GetRandomBytes` API belongs to ManagedHsm, from the official doc, but 
image
* Moreover, I've made the related tests and found that there exists errors that:
> ErrorDetails { code: Some("Unauthorized"), message: Some("Found token with aud=https://vault.azure.net, but expected aud=https://managedhsm.azure.net

These phenomenon showed me that `GetRandomBytes` is not an API belongings to `KeyClient`.

**Comment by LykxSassinator** [comment]
@heaths Done for it. FYI, I accept to make it consistent with implementations in other languages.

**Review by heaths — CHANGES_REQUESTED** [review]
Much better, thank you. One more change, though.

There are a number of APIs that don't work for both Key Vault and Managed HSM, but I'm one of the owners for the SDK and the design of both the service REST APIs and the SDKs are such that developers could test most functionality with the relatively cheap Key Vault service but deploy against a Managed HSM. Few key operations are not supported for both, including /rng.

File: sdk/security_keyvault/src/keys/models.rs [hunk 1]
```diff
@@ -403,3 +403,9 @@ pub struct EncryptResult {
     )]
     pub result: Vec<u8>,
 }
+
+#[derive(Debug, Deserialize)]
+pub struct GetRandomBytesResult {
+    #[serde(rename = "value")]
+    pub result: String,
```
[the remark this answers is not in the document]
Comment (LykxSassinator) [reply sdk/security_keyvault/src/keys/models.rs]: Got. Done for it, thx.

File: sdk/security_keyvault/src/keys/models.rs [hunk 2]
```diff
@@ -406,6 +406,10 @@ pub struct EncryptResult {
 
 #[derive(Debug, Deserialize)]
 pub struct GetRandomBytesResult {
-    #[serde(rename = "value")]
-    pub result: String,
+    #[serde(
+        rename = "value",
+        serialize_with = "ser_base64",
```
Comment (heaths) [inline sdk/security_keyvault/src/keys/models.rs:411]: It's actually base64url: https://github.com/Azure/azure-rest-api-specs/blob/be4adb7c2681e82c81c25902bb5e4964229a79a2/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.4/keys.json#L1230

**Review by heaths — CHANGES_REQUESTED** [review]

File: sdk/security_keyvault/src/keys/models.rs (hunk 2 above)
Comment (LykxSassinator) [reply sdk/security_keyvault/src/keys/models.rs]: Got it. Thx for the reminder.

File: sdk/security_keyvault/src/keys/models.rs [hunk 3]
```diff
@@ -406,10 +406,7 @@ pub struct EncryptResult {
 
 #[derive(Debug, Deserialize)]
 pub struct GetRandomBytesResult {
-    #[serde(
-        rename = "value",
-        serialize_with = "ser_base64",
-        deserialize_with = "deser_base64"
-    )]
+    /// `value` is encoded as a base64url string.
+    #[serde(rename = "value", deserialize_with = "deser_base64")]
```
Comment (heaths) [inline sdk/security_keyvault/src/keys/models.rs:410]: Ah, I see we're already doing that with this function. I forgot it was locally implemented.

**Review by heaths — APPROVED** [review]
Thank you!

**Comment by LykxSassinator** [comment]
@heaths If there was no other advices or comments, could u help me to merge it? thx.

**Comment by demoray** [comment]
Thanks for the contributions!

**Pull request merged**